lib/graph3d/StepNumber.js

```diff
@@ -115,13 +115,29 @@ StepNumber.prototype.getStep = function () {
 };
 
 /**
- * Set the current value to the largest value smaller than start, which
- * is a multiple of the step size
+ * Set the current to its starting value.
+ *
+ * By default, this will be the largest value smaller than start, which
+ * is a multiple of the step size.
+ *
+ * Parameters checkFirst is optional, default false.
+ * If set to true, move the current value one step if smaller than start.
  */
-StepNumber.prototype.start = function() {
+StepNumber.prototype.start = function(checkFirst) {
+  if (checkFirst === undefined) {
+    checkFirst = false;
+  }
+
   this._current = this._start - this._start % this._step;
+
+  if (checkFirst) {
+    if (this.getCurrent() < this._start) {
```

Comment:
...so unless the passed start value for some reason changes between the initialization of the `StepNumber` instance and the call to `start()` (which is not the case in the code), we are good.